reframe/utility/sanity.py

```diff
@@ -488,7 +488,7 @@ def assert_reference(val, ref, lower_thres=None, upper_thres=None, msg=None):
 
     if upper_thres is not None:
         try:
-            evaluate(assert_bounded(upper_thres, 0, 1))
+            evaluate(assert_bounded(upper_thres, None, None))
```

Comment:
As discussed offline, this is not enough. For positive references, the upper threshold must be in [0, inf], whereas for negative references, the lower threshold must be unbounded, [-inf, 0].